internal/pkg/cli/deploy/deploy.go

```diff
@@ -417,16 +396,30 @@ type DeployWorkloadInput struct {
 	ForceNewUpdate bool
 }
 
+// UploadArtifactsInput is the input of UploadArtifacts.
+type UploadArtifactsInput struct {
+	ImageBuilderPusher ImageBuilderPusher
+	Templater          Templater
+	Uploader           Uploader
```

Comment:
nit: can you add comments or rename the fields? I am not sure as a client it'd be clear what's the purpose of the fields

```suggestion
	ECR ImageBuilderPusher
	Addons          Templater
	S3           Uploader
```